raiden-ts/src/raiden.ts

```diff
@@ -1474,20 +1474,30 @@ export class Raiden {
             mergeMap(async (response) => response.text()),
             map((text) => {
               const suggestions = decode(SuggestedPartners, jsonParse(text));
-              assert(suggestions.length, ['empty partner suggestion from PFS', { pfs: pfs.url }]);
+
+              if (!suggestions.length) {
+                firstResponse = [];
```

Comment:
Please, set this only if `!firstResponse`, to avoid overwriting the previous error (possibly).

Reply:
The `throw` can be outside of this condition (although inside the length condition), since it's needed on both cases to continue retrying the other pfss.

Reply:
This is still an issue on latest commit.

Reply:
I think the argument from above holds here as well:
> We cannot use `(!firstResponse)` as a condition because that would mean an error from PFS1 would never be overwritten by an empty result of PFS2.